generators/client/templates/angular/src/main/webapp/app/shared/base.component.ts.ejs

```diff
@@ -0,0 +1,34 @@
+<%#
+ Copyright 2013-2020 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+import { Component, OnDestroy } from '@angular/core';
```

Comment:
Why creating base class as Angular component? I think that better is to use plain class like:
```
import { Subject } from 'rxjs';

export class SubscriptionDestroyer {
  destroy$: Subject<void> = new Subject();

  unsubscribe(): void {
    this.destroy$.next();
    this.destroy$.complete();
  }
}
```